packages/editor/src/store/reducer.js

```diff
@@ -515,6 +523,51 @@ export const editor = flow( [
 	} ),
 } );
 
+/**
+ * Reducer returning the initial edits state. With matching shape to that of
+ * `editor.edits`, the initial edits are those applied programmatically, are
+ * not considered in prmopting the user for unsaved changes, and are included
+ * in (and reset by) the next save payload.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Action object.
+ *
+ * @return {Object} Next state.
+ */
+export function initialEdits( state = INITIAL_EDITS_DEFAULTS, action ) {
+	switch ( action.type ) {
+		case 'SETUP_EDITOR':
+			if ( ! action.edits ) {
+				break;
+			}
+
+			return action.edits;
+
+		case 'SETUP_EDITOR_STATE':
```

Comment:
This is a bit confusing? Why do we need to consider two "setup" actions here? can't we just merge both in one? It probably means only use `SETUP_EDITOR_STATE` here.

Reply:
> This is a bit confusing? Why do we need to consider two "setup" actions here? can't we just merge both in one? It probably means only use `SETUP_EDITOR_STATE` here.

Right, this confused me as well, and I left it as a note in the original comment to circle back to this:

>Follow-up tasks:
>
>_[...]_
>
>`SETUP_EDITOR` handling of blocks should be eliminated in favor of a consistent, single `RESET_BLOCKS` action.

This is a bit of consistency for consistency's sake. You can see similar use in untouched existing reducers of `editor`.